package/MDAnalysis/analysis/polymer.py

```diff
@@ -0,0 +1,220 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- http://www.MDAnalysis.org
+# Copyright (c) 2006-2015 Naveen Michaud-Agrawal, Elizabeth J. Denning, Oliver Beckstein
+# and contributors (see AUTHORS for the full list)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+Polymer analysis --- :mod:`MDAnalysis.analysis.polymer`
+=======================================================
+
+
+:Author: Richard J. Gowers
+:Year: 2015
+:Copyright: GNU Public License v3
+
+This module contains various commonly used tools in analysing polymers.
+
+"""
+import numpy as np
+import logging
+
+from ..lib.util import blocks_of
+from ..lib.distances import calc_bonds
+
+logger = logging.getLogger(__name__)
+
+
+class _AnalysisBase(object):
+    """Base class for defining multi frame analysis
+
+    Defines common functions for setting up frames to analyse
+    """
+    def _setup_frames(self, **kwargs):
```

Comment:
Please use default arguments in the function definition and **kwargs only when you want to allow the caller to pass arguments to functions that you call inside your function.